### DISCUSSION TITLE
V10 Documentation Site Idea - Embeddable Live Previews with the sandpack package

### DISCUSSION THREAD

**Discussion by cliffordfajardo** [xyflow/xyflow#1789] [New Features]
I would like to suggest an idea to the community

Recently the Codesandbox team announced and open sourced the [`sandpack` - an embeddable code editor library](https://github.com/codesandbox/sandpack). This library makes it very easy to add live code samples in the browser.

This package would allow us to create live code examples in the browser like this:

```tsx
const file1 = `
 export default function App() {
      return <p> Hello,*world    </p>
  }`
`
<Sandpack
  customSetup={{ 
    dependencies: {  "react-flow-renderer": "latest" }, 
    files: {
      "/App.js": file1
  }}
/>;

// Renders an actual code editor with live preview like codesanbox.io 🪄 ✨
```
- Sandpack homepage: https://sandpack.codesandbox.io/

#### 1. For code examples on the `react-flow` website, use  [`sandpack`](https://github.com/codesandbox/sandpack) to render live examples.
  - while I was onboarding to this library, I found it quite time consuming trying to recreate the live code examples.
  - I think the documentation is good and it can be even more awesome😀 
      - Example: https://github.com/wbkd/react-flow/discussions/1778

#### 2. Create an example's gallery on the docs website: `docs_website.com/examples`
- This gallery would serve as a place for the community to share high quality examples easily (using `sandpack`) & get inspired by the library.
- When doing research on whether to adopt usage of a library or not libraries, often one of the time consuming aspects  of that  process is determining if a library is flexible enough to be used for certain use cases. An example's gallery could help here as well.
- Would likely encourage the community to give back because there is a dedicated place to share solutions more **easily**

**Reference examples of this concept used in other projects**

https://user-images.githubusercontent.com/6743796/147157169-1fba32aa-1c77-48a7-9040-b249a3fcf937.mp4

**Sandpack Example**
- [React 18 docs](https://beta.reactjs.org/learn)
- [Example blog using sandpack](https://varun.ca/modular-webgl/#data--state)

**Gallery Example Concept used in other popular projects**
- [Apache Echart's Examples Gallery](https://echarts.apache.org/examples/en/index.html)
- [AntV X6 Examples Gallery](https://x6.antv.vision/en/examples/gallery)

**Comment by moklick** [comment]
Hey @cliffordfajardo 

thanks for your input! We are moving our docs to docusaurus right now. Using sandpack instead of our custom example code sites seems to be a very good idea :) Also a gallery would be very nice..
